lib/puppet/parser/functions/consul_sorted_json.rb

```diff
@@ -1,43 +1,91 @@
 require 'json'
 
-def sorted_json(obj)
-  case obj
+module ConsulSortedJson
+
+  module_function
+
+
+  def sorted_json(config_hash, pretty)
+    cleaned = convert_integers( config_hash )
+    sorted  = sort_keys( cleaned )
+
+    if pretty
+      JSON.pretty_generate( sorted )
```

Comment:
You are doing post-processing (making is pretty) after sorting (line `10`). How are you ensuring `pretty_generate` and `generate` will still keep the keys in order? As you might already know, ruby does *not* guarantee keys order once you serialize a hash to JSON.

http://stackoverflow.com/questions/5517007/json-parse-changes-sorting-ruby/5518671#5518671